var/spack/repos/builtin/packages/py-pybind11/package.py

```diff
@@ -46,7 +46,9 @@ class PyPybind11(CMakePackage):
     version('2.1.1', '5518988698df937ccee53fb6ba91d12a')
     version('2.1.0', '3cf07043d677d200720c928569635e12')
 
-    depends_on('py-pytest', type=('build'))
+    variant('test', default=False, description="Builds tests")
```

Comment:
ideally this should not be a variant, but rather controlled via `install` parameter:
```
  --run-tests           run package tests during installation (same as --test=all)
```

However, I don't think `depends_on` support this  😞

Reply:
It does, we now have a `type='test'`.

Reply:
How would that work with setting `-DPYBIND11_TEST:BOOL=ON` or `-DPYBIND11_TEST:BOOL=OFF` in the `install` function?